statemachine/src/main/java/com/amazonaws/stepfunctions/cloudformation/statemachine/ResourceHandler.java

```diff
@@ -17,7 +17,9 @@
     protected ProgressEvent<ResourceModel, CallbackContext> handleDefaultError(ResourceHandlerRequest<ResourceModel> request, Exception e, MetricsRecorder metricsRecorder) {
         ProgressEvent.ProgressEventBuilder<ResourceModel, CallbackContext> resultBuilder = ProgressEvent.<ResourceModel, CallbackContext>builder();
 
-        metricsRecorder.setMetricsFromException(e);
+        if (metricsRecorder != null) {
+            metricsRecorder.setMetricsFromException(e);
+        }
```

Comment:
what's the circumstance where metricsRecorder is null?

Reply:
I've removed metricsRecorder from the deleteHandler code, as it complicates the logging when the handler is being invoked multiple times for the same initial operation. We were only tracking the delete success, which is already provided as a default metric.